rules/danglingwhitespace/rule.go

```diff
@@ -0,0 +1,49 @@
+package danglingwhitespace
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/vbatts/git-validation/git"
+	"github.com/vbatts/git-validation/validate"
+)
+
+var (
+	// DanglingWhitespace is the rule for checking the presence of dangling
+	// whitespaces on line endings.
+	DanglingWhitespace = validate.Rule{
+		Name:        "dangling-whitespace",
+		Description: "checking the presence of dangling whitespaces on line endings",
+		Run:         ValidateDanglingWhitespace,
+	}
+)
+
+func init() {
+	validate.RegisterRule(DanglingWhitespace)
+}
+
+func ValidateDanglingWhitespace(c git.CommitEntry) (vr validate.Result) {
+	diff, err := git.Show(c["commit"])
```

Comment:
If you don't save the diff for reuse between multiple validators, I think it would be faster (no need to serialize the whole diff) and less work in this repo (no need to parse the diff or formulate an error message) to use `git show --check {commit}` (like opencontainers/runtime-spec@bdaf4467e, .tools: Run `git show --check ...` on each commit, 2015-09-12, opencontainers/runtime-spec#182).